Geometry_Engine/Query/Width.cs

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.ComponentModel;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Query
+    {
+
+        /***************************************************/
+        /****       Public Methods - BoundingBox        ****/
+        /***************************************************/
+
+        [Description("Returns the horizontal orthogonal width of a BHoM BoundingBox.")]
+        [Input("boundingBox", "BHoM BoundingBox to query its width.")]
+        [Output("width", "The horizontal width of the BoundingBox based on the difference in XY values for its bounding box")]
+        public static double Width(this BoundingBox boundingBox)
```

Comment:
Don't think Width is the right name for this method. Would call it something like `HorizontalDiagonalLength` or similar. Does not need to be exactly that, but needs to be something that states what the method is returning.